### ISSUE TITLE
Feat: Implement Chat Interface with MCP-Server for Kubernetes Intelligent

### ISSUE THREAD

**Issue by elliotxx** [KusionStack/karpor#658]
## What would you like to be added?
Add an intuitive chat interface on the Karpor homepage that leverages MCP (Model Context Protocol) to interact with Karpor Storage for Kubernetes Intelligent.

## Why is this needed?
This feature will provide users with a natural language interface to understand and manage their Kubernetes clusters and resources. The chat interface will:
    - Connect to MCP Server for standardized AI interactions
    - Integrate with Karpor Storage for comprehensive cluster data access
    - Support natural language queries about cluster status and resources
    - Provide real-time cluster insights and recommendations
    
    Benefits:
    - Simplified cluster insight through natural language
    - Implementing MCP server to achieve standard communication with LLM
    - Better understanding of cluster state through interactive queries
    - Streamlined troubleshooting with intelligent assistance
    - Enhanced user experience for both novice and expert users

**Comment by rajp152k** [comment]
inserting refs: https://github.com/metoro-io/mcp-golang  
am interested in collaborating over this

**Comment by elliotxx** [comment]
@rajp152k Yeahh, recently I have also paid attention to mcp-golang in [this article](https://dev.to/chrisbattarbee/write-tools-for-llms-mcp-golang-2nmf), and I have high hopes for what this Issue can do. In fact, I am currently developing other features of [v0.6.0](https://github.com/KusionStack/karpor/milestone/5) and cannot escape from it temporarily.
If you are interested in this Issue, it's cool, but I am happy to hand over it to the brother of the community!

**Comment by elliotxx** [comment]
And if you need the front-end development, another guy can help @hai-tian

**Comment by rajp152k** [comment]
I see, I am also building up my knowledge of karpor and the kusion ecosystem because I would like to better grasp the big picture..

definitely see you in multiple issues here and there..

as for this issue, I'm investing effort and will keep you guys posted.

**Comment by elliotxx** [comment]
@rajp152k OK, please feel free to harass me. I'd be happy to participate in the discussion of mcp. If you have any questions about kusion, you can also at @ffforest

**Comment by rajp152k** [comment]
created #675

**Comment by rajp152k** [comment]
@elliotxx , PR #795 , initializes the development process as last discussed in #675

As you suggested we should proceed with an SSE server rather than an stdio one, I thought it would be best to have it as another subcommand of karpor (like syncer is right now)

I've added the initial code right now with directives as to how I'm planning to proceed in the comments : 
 - the elastic storage code in the /pkg/infra/search/storage/elasticsearch directory
 - I'm planning on using mcp-golang instead of mcp-go now : the former seems fully featured and more stable
    - mcp-go only exposes complete functionality for stdio servers but mcp-golang does so for SSE servers as well

Please let me know what you think

**Comment by elliotxx** [comment]
@rajp152k Thank you for your quick action! Give me some time to take a look.

**Comment by elliotxx** [comment]
@rajp152k I leave some comments, ptal #795

**Comment by rajp152k** [comment]
@elliotxx , have incorporated the updates

**Comment by rajp152k** [comment]
@elliotxx , please see the final commit of #798,
there were some issues with the golangci-lint config   
have made some changes (comments in the commit) but need to be reviewed if they conform to the whole project

**Comment by elliotxx** [comment]
@rajp152k Yeah, the modification itself is not a problem, but I found other lint issues while running `make lint` in the local regression test

![Image](https://github.com/user-attachments/assets/887cb6be-1512-467b-9e13-31517f473b74)

**Comment by rajp152k** [comment]
Can you pull and check the latest branch, I've fixed these issues with my local iterations

**Comment by elliotxx** [comment]
> Can you pull and check the latest branch, I've fixed these issues with my local iterations

@rajp152k  Yeah, I checkout locally to your branch in local, and now it's fine.

**Comment by elliotxx** [comment]
Yeah 👍, I leave comment in PR
